### PR TITLE
Validation block scheduler

### LINKED ISSUE #236: Scheduler for validation blocks
Validation blocks need special treatment in the scheduler. Each validator seat should be entitled to a fixed throughput per slot, completely isolated from the main scheduler and from the throughput of other validation blocks.

### LINKED ISSUE #338: Penalize over-issuing validators
Three options

1. Blacklisting: if a validator over-issues within a slot, this is trivial to detect once that slot is committed by simply counting the number of accepted blocks from the validator in that slot. If they overissue, we mark this validator as blacklisted for the epoch in the accounts Manager. Then, similarly to how we handle block issuers with negative BIC, we can filter all blocks from blacklisted validators based on the objective information in the account manager.

2. Charge Mana: if the validators over-issue, we can charge them Mana for any blocks in excess of the desired number. This is accounted for upon commitment so it is easy to count the number of additional blocks and simply subtract a Mana penalty from the validator’s BIC. If they have not alloted Mana to themselves, this will result in their BIC going negative and effectively becoming blacklisted as well. 

3. Reduce rewards: we can take the over-issuance into account in our calculation of the performance factor.

### PR DESCRIPTION
**Pull request by cyberphysic4l**
Implementation of the validation block scheduler described in [this Google doc](https://docs.google.com/document/d/1trytLYpPyfvU-GaA4DCS4uhOckkl1-3Cxe1RrkPzRD8/edit).

Inludes fixed rate of `CommitteeSize` x `ValidationBlocksPerSlot` and for each validator.  penalty of `PunishmentEpochs` epochs worth of Mana generation from the validator's stake for each block issued over the `ValidationBlocksPerSlot`. 

closes #236 

closes #338

### CODE REVIEW

File: pkg/protocol/engine/congestioncontrol/scheduler/drr/validatorqueue.go
Comment (piotrm50) [inline pkg/protocol/engine/congestioncontrol/scheduler/drr/validatorqueue.go:106]: This method is not used. I think ValidatorQueue should have a limited size and the tail should be removed if that is exceeded, just like for regular blocks.

**Review by piotrm50 — CHANGES_REQUESTED** [review]

File: pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go
Comment (cyberphysic4l) [reply pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go:288]: Yes, we need to mark validation blocks as ready just like any other block. Is this the part you are referring to?

File: pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go
Comment (cyberphysic4l) [reply pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go:165]: Yes, I think it would be good to have a metric for each validator queue separately.

Comment (cyberphysic4l) [reply pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go:160]: done.

File: pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go
Comment (piotrm50) [reply pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go]: I meant the commented out line.

File: pkg/protocol/engine/filter/blockfilter/filter.go
Comment (piotrm50) [inline pkg/protocol/engine/filter/blockfilter/filter.go:90]: Wait, I thought that candidate validators should always issue validation blocks as well, so we can't issue them here. Or am I missing something?

**Review by piotrm50 — COMMENTED** [review]
Looks good, apart from one question I had about changes to the filter. I will approve after clarifying that. 

Before merging please make sure that it runs in Docker network correctly.

File: pkg/protocol/engine/filter/blockfilter/filter.go
Comment (cyberphysic4l) [reply pkg/protocol/engine/filter/blockfilter/filter.go:90]: non-committee validators must issue basic blocks using their own Mana in order to put themselves forward for committee selection. Validation blocks are now reserved only for committee.

**Review by piotrm50 — APPROVED** [review]

**Pull request merged**
